internal/services/eventhub/eventhub_resource.go

```diff
@@ -173,16 +172,52 @@ func resourceEventHubCreateUpdate(d *pluginsdk.ResourceData, meta interface{}) e
 
 	id := eventhubs.NewEventhubID(subscriptionId, d.Get("resource_group_name").(string), d.Get("namespace_name").(string), d.Get("name").(string))
 
-	if d.IsNewResource() {
-		existing, err := client.Get(ctx, id)
+	partitionCount := int64(d.Get("partition_count").(int))
+	messageRetention := int64(d.Get("message_retention").(int))
+	eventhubStatus := eventhubs.EntityStatus(d.Get("status").(string))
+
+	parameters := eventhubs.Eventhub{
+		Properties: &eventhubs.EventhubProperties{
+			PartitionCount:         &partitionCount,
+			MessageRetentionInDays: &messageRetention,
+			Status:                 &eventhubStatus,
+		},
+	}
```

Comment:
we can combine this?
```suggestion
	parameters := eventhubs.Eventhub{
		Properties: &eventhubs.EventhubProperties{
			PartitionCount:         int64(d.Get("partition_count").(int)),
			MessageRetentionInDays: int64(d.Get("message_retention").(int)),
			Status:                 eventhubs.EntityStatus(d.Get("status").(string)),
		},
	}
```